src/conn/conn_evict.c

```diff
@@ -0,0 +1,380 @@
+    if (evict->eviction_updates_trigger > evict->eviction_trigger) {
+        WT_CONFIG_DEBUG(session,
+          "config eviction_updates_trigger=%f cannot exceed eviction_trigger=%f. Setting "
+          "eviction_updates_trigger to %f.",
+          evict->eviction_updates_trigger, evict->eviction_trigger, evict->eviction_trigger);
+        evict->eviction_updates_trigger = evict->eviction_trigger;
+    }
+
+    /* The target size must be lower than the trigger size or we will never get any work done. */
+    if (evict->eviction_target >= evict->eviction_trigger)
+        WT_RET_MSG(session, EINVAL, "eviction target must be lower than the eviction trigger");
+    if (evict->eviction_dirty_target >= evict->eviction_dirty_trigger)
+        WT_RET_MSG(
+          session, EINVAL, "eviction dirty target must be lower than the eviction dirty trigger");
+    if (evict->eviction_updates_target >= evict->eviction_updates_trigger)
+        WT_RET_MSG(session, EINVAL,
+          "eviction updates target must be lower than the eviction updates trigger");
+
+    return (0);
+}
+
+/*
+ * __evict_config_local --
+ *     Configure eviction.
+ */
+static int
+__evict_config_local(WT_SESSION_IMPL *session, const char *cfg[])
+{
+    WT_CONFIG_ITEM cval;
+    WT_CONNECTION_IMPL *conn;
+    WT_EVICT *evict;
+    uint32_t evict_threads_max, evict_threads_min;
+
+    conn = S2C(session);
+    evict = conn->evict;
+
+    WT_RET(__validate_evict_config(session, cfg));
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction.threads_max", &cval));
+    WT_ASSERT(session, cval.val > 0);
+    evict_threads_max = (uint32_t)cval.val;
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction.threads_min", &cval));
+    WT_ASSERT(session, cval.val > 0);
+    evict_threads_min = (uint32_t)cval.val;
+
+    if (evict_threads_min > evict_threads_max)
+        WT_RET_MSG(
+          session, EINVAL, "eviction=(threads_min) cannot be greater than eviction=(threads_max)");
+    conn->evict_threads_max = evict_threads_max;
+    conn->evict_threads_min = evict_threads_min;
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction.evict_sample_inmem", &cval));
+    conn->evict_sample_inmem = cval.val != 0;
+
+    /* Retrieve the wait time and convert from milliseconds */
+    WT_RET(__wt_config_gets(session, cfg, "cache_max_wait_ms", &cval));
+    evict->cache_max_wait_us = (uint64_t)(cval.val * WT_THOUSAND);
+
+    /* Retrieve the timeout value and convert from seconds */
+    WT_RET(__wt_config_gets(session, cfg, "cache_stuck_timeout_ms", &cval));
+    evict->cache_stuck_timeout_ms = (uint64_t)cval.val;
+
+    return (0);
+}
+
+/*
+ * __wti_eviction_config --
+ *     Configure or reconfigure eviction.
+ */
+int
+__wti_eviction_config(WT_SESSION_IMPL *session, const char *cfg[], bool reconfig)
+{
+    WT_CONNECTION_IMPL *conn;
+
+    conn = S2C(session);
+
+    WT_ASSERT(session, conn->evict != NULL);
+
+    WT_RET(__evict_config_local(session, cfg));
```

Comment:
The `evict_config_local` function is quite small. And this is the *only* caller of it. Why is it a function? Why not have that code in here directly given its small size?